### PR TITLE
crypto/bn: remove redundant parenthesis

### PR DESCRIPTION
**Pull request by vszakats**

### CODE REVIEW

File: crypto/bn/bn_asm.c [hunk 1]
```diff
@@ -210,10 +210,10 @@ BN_ULONG bn_div_words(BN_ULONG h, BN_ULONG l, BN_ULONG d)
     int i, count = 2;
 
     if (d == 0)
-        return (BN_MASK2);
+        return BN_MASK2;
 
     i = BN_num_bits_word(d);
-    assert((i == BN_BITS2) || (h <= (BN_ULONG)1 << i));
+    assert(i == BN_BITS2 || (h <= (BN_ULONG)1 << i));
```
[the remark this answers is not in the document]
Comment (vszakats) [reply crypto/bn/bn_asm.c]: Added a commit with a few more. Including the above, plus some trivial ones missed in the second commit. (there are some more that could be removed, but it's probably okay to let a few ones there to help readability.)

**Comment by t-j-h** [comment]
Redundant parentheses help readability - this whole set of changes is not something I think makes sense to put in place.

**Comment by vszakats** [comment]
This sounds quite confusing after a large number of commits exactly like having been committed to the repository. I'd argue that `return (x);` ruins readability compared to `return x;`. My first two commits fall into the trivially unnecessary parenthesis, the third one, which I added based on @levitte's comments, are less trivial.

I'd prefer a maintainer clarify this to avoid unnecessary work and frustration.

**Comment by vszakats** [comment]
Latest commit with similar content was merged in the last few days: https://github.com/openssl/openssl/commit/8686c474807d3c7bcb722294f39ac4e5627e9fa2

**Comment by t-j-h** [comment]
You've mixed in both changing parentheses around return statement with other items.
Our coding style doesn't have a fixed view on this - reflecting the team having varying views on the topic.
It is a style issue - and you will get different view points if it is not something covered in the coding style representing an agreed team view.

**Comment by vszakats** [comment]
In the first commit only returns were touched, the second one removed clearly unnecessary ones; there wasn't many of those and the rest of the existing code didn't use parents in similar locations. Existing code also uses inconsistent style for the same idioms, so it seems there is no consistent style even inside a single source code. This, @levitte and your comment gives completely contradicting signals, so I'm closing this as probably my last attempt to contribute to this project. From a potential contributor POV it'd be useful to advertise a single style and stick to it, this way it's just time waste.

**Comment by levitte** [comment]
Errrr, I don't see a need to close this.  Sorry that you got your feathers ruffled, @vszakats, but we do allow openly different opinion and discussion around them in this project.

That being said, the removal of extra parentheses around `return` values has been going on for quite some time and without discussion, by different people as part of whatever PR they presented, so I don't think @t-j-h meant those specifically.  It might even be that @t-j-h's comment was partly directed at me for my comments about removing even more parens.

(and @t-j-h, @vszakats *did* split this PR into two commits, precisely to separate `return` cases from the rest)

**Comment by t-j-h** [comment]
This is a *mix* of return parentheses removal (a pure style issue) and removal of additional parentheses which improve readability (although the views on that vary in the team). And yes @levitte you did encourage @vszakats to go beyond the simple return removal.

**Comment by levitte** [comment]
I'm coming back to this, because I'm not noticing additional commits that weren't there when I originally looked at this.  You might want to consider separate PRs, 'cause as it is now, it's getting increasingly difficult to approve everything in a bulk.

That is, if you decide to come back.  Until then, I think I'm done here.

**Comment by vszakats** [comment]
This isn't a mix, I intentionally created separate commits for each cleanup pass. One for returns, one for trivially redundant parenthesis and an additional one with less trivial cases (which included a few trivial ones I've just found in this pass), plus it includes two more commit involving comment (one typo and function name markup which I picked up from past commits made by others - maybe wrongly) and URL cleanups. These commits can be shuffled/dropped/recreated freely and separately before a merge.

Last thing I want is a style debate, so I go for the indisputable low-hanging opportunities in the hope these can be accepted without controversy or long arguments.

I was prepared to do the same for the other 60 modules with similar issues, but it seems that even going with the trivial is a ground for debate.

**Comment by vszakats** [comment]
I have no way to tell what should be put in separate PRs, maybe comment/URL and parenthesis, but which ones? Maybe the best is if you cherry-pick the specific commits you guys agree with.

**Comment by vszakats** [comment]
Forgot to reopen before force push, so opened a new PR with these same changes, save the possibly disputed hunks from commit `crypto/bn: remove redundant parenthesis (more)`: https://github.com/openssl/openssl/pull/4273

**Comment by richsalz** [comment]
This is the internet, there's lots of discussion and debate around everything :)
Put the retnr()->return in one PR; put the other paren changes in separate PR's.
The PR is our unit of approval, not individual commits.

**Pull request closed without merging**